pilopt/src/lib.rs

```diff
@@ -480,23 +481,42 @@ fn constrained_to_constant<T: FieldElement>(
     None
 }
 
-/// Removes identities that evaluate to zero and lookups with empty columns.
+/// Removes identities that evaluate to zero, the same column and lookups with empty columns.
```

Comment:
```suggestion
/// Removes identities that evaluate to zero and lookups with empty columns.
```
(I think one can argue that `X - X` evaluates to zero :)

Reply:
Maybe you can add `(including constraints of the form "X = X")`